### DISCUSSION TITLE
New server installation is 2 times slower

### DISCUSSION THREAD

**Discussion by TomLorenzi** [symfony/symfony#49084] [Q&A]
Hey,

I just installed a new server and setup my symfony project on it.  
The problem is that now my projet is 2 times slower than all the previous installations I've done so far.

I can't figure out how and why because I'm literally using the same php version / curl version / extensions and so same php.ini conf.

I can provide blackfire profiles but I've been digging in these for many times and can't find any clue.

If someone have any idea of what I could do I would be glad !

New and slow project : https://blackfire.io/profiles/ca036c9b-bd89-4483-b7cd-0ae28faf098a/graph

Old and fast project : https://blackfire.io/profiles/5815cece-d05c-489f-82b1-8c42c24419cd/graph

Using php 7.4 and Symfony 5.4 and new server should be more performant than the older one (better CPU / RAM / bandswitch / ...)

**Comment by MatTheCat** [comment]
Ideally you would compare execution on production environments without IO, as most of the time is spent in your `ApiRequestService`.

Then we could investigate a potential slowdown.

**Reply by TomLorenzi** [reply]
@MatTheCat is there a parameter that I can't find in blackfire that allows to ignore IO, or i should just return empty response instead of my api requests ?

**Reply by MatTheCat** [reply]
Not that I'm aware of 🤔

Don't you have some pages not requiring IO? Also make sure caches are warmed up before profiling so no time is spent refreshing them.

**Reply by TomLorenzi** [reply]
I could provide you a profile of the api that doesn't do any request maybe it will be better for you

**Reply by TomLorenzi** [reply]
@MatTheCat Maybe it this will be more usefull to you ?
I really can't figure out what is problematic here :/

Old and fast : https://blackfire.io/profiles/fc0a0059-83df-4389-8506-1e2790a3270f/graph
New and slow : https://blackfire.io/profiles/ad02f736-5efb-4366-aaa1-fd726fb44161/graph

And if you want to see the comparaison : https://blackfire.io/profiles/compare/fc692665-58f4-4d4a-91b3-6e0bf3630aa7/graph

**Reply by MatTheCat** [reply]
The comparison says `ObjectHydrator::hydrateRowData` is called 409 vs 157 times so I guess the difference only comes from the volume of data being handled.

**Reply by TomLorenzi** [reply]
+400% seems such a big difference, there's some data difference but it's almost the same, it really feels really slow here and if i take an example with less data on the new and slow server it still remain slower (both link from wrong server): 

https://blackfire.io/profiles/4558256e-4121-4d94-9b36-d69499807a19/graph

https://blackfire.io/profiles/0492e9c9-de23-413d-a8a5-6ef46e15c5c4/graph

Something just really feel wrong here 😞

**Reply by MatTheCat** [reply]
> +400% seems such a big difference, there's some data difference but it's almost the same

The comparison says +201% and you had +261% data rows so it seems rather logical.

**Reply by TomLorenzi** [reply]
Then why would it be still slower with less rows as provided with my last example ?

**Reply by MatTheCat** [reply]
In your last post you wrote `both link from wrong server` and I don’t know what it’s supposed to mean :thinking:

**Reply by TomLorenzi** [reply]
Oh sorry it wasn't that easy to understand, both profile came from the new and slow server, and here I have less data but it's still slower.

**Reply by MatTheCat** [reply]
The “slow” profile is caching data.

> make sure caches are warmed up before profiling so no time is spent refreshing them.

**Reply by TomLorenzi** [reply]
So last question, but this would only be result as hardware performance ?  (knowing that the slowest got better hardware :/)

Slow : https://blackfire.io/profiles/4ba54521-165f-49a1-94cb-3b3e36d1a63e/graph
Fast : https://blackfire.io/profiles/133eea1f-a626-4d80-9fbc-63fe76f95b4a/graph

I'll continue digging into this, maybe not symfony fault

**Reply by MatTheCat** [reply]
Your last profiles show ≈ 2s difference between 10 000  `array_unshift` executions so if there is an issue it does not come from Symfony!

As a last idea, did you make sure OPcache is enabled and used on both servers?

**Reply by TomLorenzi** [reply]
Yes OPCache is enabled on both servers,
Well i keep looking in my server configuration but something is defintly wrong, thank you !
